---
layout: post
title: Hi-C analysis of Drosophila melanogaster cells using HiCExplorer
date: 2020-08-09 01:29:21 +0230
description: Youâ€™ll find this post in your `_posts` directory. Go ahead and edit it and re-build the site to see your changes. # Add post description (optional)
img: # Add image post (optional)
fig-caption: # Add figcaption (optional)
tags: [Bioinformatics, Epigenetics]
---


<h2> Introduction </h2>

<p align="justify"> The <a href = "https://github.com/deeptools/HiCExplorer" > HiCExplorer </a> is available from the repository <b>suite_hicexplorer</b>. It is a set of tools from <i>deeptools</i>, dedicated towards analysis of data from the 3D conformation of the chromatin.</p>

<ol>
<li>Map the Hi-C reads to the reference genome</li>
<li>Creation of a Hi-C matrix</li>
<li>Plotting the Hi-C matrix</li>
<li>Correction of Hi-C matrix</li>
<li>TAD Calling</li>
<li>A/B compartments computation</li>
<li>pyGenomeTracks visualization</li>
<li>Loop detection</li>
</ol>


<h2> References </h2>
<ol>
<li> Joachim Wolff, Fidel Ramirez, Vivek Bhardwaj, Ekaterina Polkh, 2020 Hi-C analysis of Drosophila melanogaster cells using HiCExplorer (Galaxy Training Materials). /training-material/topics/epigenetics/tutorials/hicexplorer/tutorial.html Online; accessed Sat Aug 08 2020 </li>
<li> Batut et al., 2018 Community-Driven Data Analysis Training for Biology Cell Systems 10.1016/j.cels.2018.05.012  </li>
</ol>
